spec/functions/fetch_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'spec_helper'
+
```

Comment:
Please restructure this to match the current pattern of unit tests in stdlib.

Using the `run.with_params(...).and_return(...)` matcher makes for so much nicer reading of the code.